### DISCUSSION TITLE
Can't Add External Plugin

### DISCUSSION THREAD

**Discussion by james9078** [sofa-framework/sofa#3733] [Build / Config / Environment] [answered]
Hi,

I am trying to add an external plugin with SOFA v22.12, following this instruction https://www.youtube.com/watch?v=46E215871e8, but keep running into this error:
CMake Error at //path/to/SOFA/ext_plugin_repo/CMakeLists.txt:5 (sofa_add_subdirectory):
  sofa_add_subdirectory Macro invoked with incorrect arguments for macro
  named: sofa_add_subdirectory

This is what I have in the CMakeLists.txt file:
cmake_minimum_required(VERSION 2.8.12)

find_package(SofaFramework)

sofa_add_subdirectory(MyAwesomeComponents/ MyAwesomeComponents)

I have also tried specifying the version of the project: sofa_add_subdirectory(MyAwesomeComponents/ MyAwesomeComponents VERSION 1.0). 

Thank you.

**Comment by alxbilger** [comment, marked as answer]
Try `sofa_add_subdirectory(plugin MyAwesomeComponents MyAwesomeComponents)`. I think this function changed since the video.

**Reply by james9078** [reply]
Thank you sir!

**Reply by zhanglepy** [reply]
This method is effective for Sofa v22.12 version, but not for Sofa v21.12 version. What should I do?

**Comment by zhanglepy** [comment]
This method is effective for Sofa v22.12 version, but not for Sofa v21.12 version. What should I do?
